File: dart/structural/composite/graphic/bin/circle.dart
import './dot.dart';

class Circle extends Dot {
  double radius;

  Circle(int x, int y, double radiusValue) : super(x, y) {
    radius = radiusValue;
  }

  @override
  void draw() {
    print('CIRCLE: x = $x, y = $y, radius = $radius');
  }
}

File: typescript/structural/composite/graphic/circle.ts
import {Dot} from "./dot";

export class Circle extends Dot {
  radius: number

  constructor(x: number, y: number, radius: number) {
    super(x, y);
    this.radius = radius
  }

  draw() {
    super.draw();
    console.log(`CIRCLE: radius = ${this.radius}`)
  }
}
